server/sonar-server/src/test/java/org/sonar/server/computation/step/ComputeAndPersistDaysSinceLastCommitStepTest.java

```diff
@@ -0,0 +1,137 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.computation.step;
+
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.sonar.api.utils.DateUtils;
+import org.sonar.api.utils.System2;
+import org.sonar.batch.protocol.Constants;
+import org.sonar.batch.protocol.output.BatchReport;
+import org.sonar.batch.protocol.output.BatchReportReader;
+import org.sonar.batch.protocol.output.BatchReportWriter;
+import org.sonar.core.component.ComponentDto;
+import org.sonar.core.measure.db.MetricDto;
+import org.sonar.core.persistence.DbTester;
+import org.sonar.server.computation.ComputationContext;
+import org.sonar.server.computation.measure.MetricCache;
+import org.sonar.server.db.DbClient;
+import org.sonar.server.measure.persistence.MeasureDao;
+import org.sonar.server.source.index.SourceLineIndex;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ComputeAndPersistDaysSinceLastCommitStepTest extends BaseStepTest {
+
+  @ClassRule
+  public static DbTester db = new DbTester();
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+  File dir;
+
+  ComputeAndPersistDaysSinceLastCommitStep sut;
+
+  DbClient dbClient;
+  SourceLineIndex sourceLineIndex;
+  MetricCache metricCache;
+
+  @Before
+  public void setUp() throws Exception {
+    dbClient = new DbClient(db.database(), db.myBatis(), new MeasureDao());
+    sourceLineIndex = mock(SourceLineIndex.class);
+    metricCache = mock(MetricCache.class);
+    when(metricCache.get(anyString())).thenReturn(new MetricDto().setId(10));
+    dir = temp.newFolder();
+    db.truncateTables();
+
+    sut = new ComputeAndPersistDaysSinceLastCommitStep(System2.INSTANCE, dbClient, sourceLineIndex, metricCache);
+  }
+
+  @Override
+  protected ComputationStep step() throws IOException {
+    return sut;
+  }
+
+  @Test
+  public void persist_number_of_days_since_last_commit_from_report() throws Exception {
+    long threeDaysAgo = DateUtils.addDays(new Date(), -3).getTime();
+    BatchReportWriter reportWriter = initReportWithProjectAndFile();
+    reportWriter.writeComponentScm(
+      BatchReport.Scm.newBuilder()
+        .setComponentRef(2)
+        .addChangeset(
+          BatchReport.Scm.Changeset.newBuilder()
+            .setDate(threeDaysAgo)
+        )
+        .build()
+      );
+    ComputationContext context = new ComputationContext(new BatchReportReader(dir), new ComponentDto().setId(456_789L).setUuid("project-uuid"));
```

Comment:
Setting the id is useless. I would also use ComponentTesting.newProject("project-uuid")